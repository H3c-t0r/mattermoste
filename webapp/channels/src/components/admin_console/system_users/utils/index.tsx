// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {FormattedMessage} from 'react-intl';

import {UserReportSortColumns, ReportSortDirection} from '@mattermost/types/reports';
import type {UserReportOptions, UserReport} from '@mattermost/types/reports';

import {PAGE_SIZES, type SortingState} from 'components/admin_console/list_table';

import type {AdminConsoleUserManagementTableProperties} from 'types/store/views';

import {ColumnNames, RoleFilters, StatusFilter} from '../constants';
import type {TableOptions} from '../system_users';

export function convertTableOptionsToUserReportOptions(tableOptions?: TableOptions): UserReportOptions {
    return {
        page_size: tableOptions?.pageSize || PAGE_SIZES[0],
        from_column_value: tableOptions?.fromColumnValue,
        from_id: tableOptions?.fromId,
        direction: tableOptions?.direction,
        ...getSortColumnForOptions(tableOptions?.sortColumn),
        ...getSortDirectionForOptions(tableOptions?.sortIsDescending),
        search_term: tableOptions?.searchTerm,
        ...getUserStatusFilterOption(tableOptions?.filterStatus),
        ...getRoleFilterOption(tableOptions?.filterRole),
    };
}

/**
 * Converts the sorting column name to API compatible sorting column name. Default sorting column name is by username.
 */
export function getSortColumnForOptions(id?: SortingState[0]['id']): Pick<UserReportOptions, 'sort_column'> {
    let sortColumn: UserReportOptions['sort_column'];

    if (id === ColumnNames.email) {
        sortColumn = UserReportSortColumns.email;
    } else if (id === ColumnNames.createAt) {
        sortColumn = UserReportSortColumns.createAt;
    } else {
        // Default sorting to first User details column
        sortColumn = UserReportSortColumns.username;
    }

    return {
        sort_column: sortColumn,
    };
}

/**
 * Converts the sorting direction to API compatible sorting direction. Default sorting direction is ascending.
 */
export function getSortDirectionForOptions(desc?: SortingState[0]['desc']): Pick<UserReportOptions, 'sort_direction'> {
    let sortDirection: UserReportOptions['sort_direction'];

    if (desc) {
        sortDirection = ReportSortDirection.descending;
    } else {
        sortDirection = ReportSortDirection.ascending;
    }

    return {
        sort_direction: sortDirection,
    };
}

/**
 * It returns the value of that column on which sorting is applied.
 */
export function getSortableColumnValueBySortColumn(row: UserReport, sortColumn: AdminConsoleUserManagementTableProperties['sortColumn']): string {
    switch (sortColumn) {
    case ColumnNames.email:
        return row.email;
    case ColumnNames.createAt:
        return String(row.create_at);
    default:
        return row.username;
    }
}

export function getUserStatusFilterOption(status?: string): Partial<Pick<UserReportOptions, 'hide_active' | 'hide_inactive'>> {
    if (status === StatusFilter.Active) {
        return {
            hide_inactive: true,
        };
    } else if (status === StatusFilter.Deactivated) {
        return {
            hide_active: true,
        };
    }

    return {};
}

export function getPaginationInfo(pageIndex: number, pageSize: number, currentLength: number, total?: number) {
    if (!currentLength) {
        return (
            <FormattedMessage
                id='admin.system_users_list.pagination.no_users'
                defaultMessage='0 users'
            />
        );
    }

    const firstPage = (pageIndex * pageSize) + 1;
    const lastPage = (pageIndex * pageSize) + currentLength;
    const totalItems = total || 0;

    return (
        <FormattedMessage
            id='admin.system_users_list.pagination'
            defaultMessage='Showing {firstPage} - {lastPage} of {totalItems} users'
            values={{
                firstPage,
                lastPage,
                totalItems,
            }}
        />
    );
}

export function getDefaultValueFromList<T extends {value: string}>(value: string, options: T[]) {
    const option = options.find((option) => option.value === value);

    if (option) {
        return option;
    }

    return options[0];
}

export function getRoleFilterOption(role?: string): Pick<UserReportOptions, 'role_filter'> {
    if (!role || role === RoleFilters.Any) {
        return {role_filter: ''};
    }
    return {role_filter: role};
}
