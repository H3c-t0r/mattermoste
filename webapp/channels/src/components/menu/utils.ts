// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Children, ReactNode, cloneElement, isValidElement} from 'react';

import {MENU_ITEM_KEY_PREFIX, Props as MenuItemProps} from './menu_item';
import {SUB_MENU_ITEM_KEY_PREFIX} from './sub_menu';

export function injectPropsInMenuItems(children: ReactNode[], forceCloseMenu?: MenuItemProps['forceCloseMenu'], forceCloseSubMenu?: MenuItemProps['forceCloseSubMenu']) {
    const modifiedChildren = Children.map(children, (child) => {
        if (!isValidElement(child)) {
            return null;
        }

        if (
            child.props &&
            child.props.id &&
            (child.props.id.startsWith(MENU_ITEM_KEY_PREFIX) || child.props.id.startsWith(SUB_MENU_ITEM_KEY_PREFIX))
        ) {
            return cloneElement(child, {
                forceCloseMenu,
                forceCloseSubMenu,
            } as MenuItemProps);
        }

        return child;
    });

    return modifiedChildren;
}

export function createMenusUniqueId(prefix: string, menuItemName: string, ...uniqueValues: string[]) {
    const uniqueValuesJoined = uniqueValues.join('-');
    return `${prefix}__${menuItemName}__${uniqueValuesJoined}`;
}
