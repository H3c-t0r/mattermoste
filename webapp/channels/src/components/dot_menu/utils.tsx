// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {trackEvent} from 'actions/telemetry_actions';

import {EventTypes, TELEMETRY_CATEGORIES} from 'utils/constants';

import type React from 'react';

export type ChangeEvent = React.KeyboardEvent | React.MouseEvent;

export const trackDotMenuEvent = (e: ChangeEvent, suffix: string): void => {
    if (e.type === EventTypes.CLICK) {
        trackEvent(TELEMETRY_CATEGORIES.POST_INFO_MORE, EventTypes.CLICK + '_' + suffix);
    } else {
        trackEvent(TELEMETRY_CATEGORIES.POST_INFO_MORE, EventTypes.SHORTCUT + '_ ' + suffix);
    }
};
