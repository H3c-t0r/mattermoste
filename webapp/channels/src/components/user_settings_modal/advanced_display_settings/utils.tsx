// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {defineMessage, FormattedMessage} from 'react-intl';

import type {FieldsetCheckbox} from 'components/widgets/modals/components/checkbox_setting_item';
import type {FieldsetRadio} from 'components/widgets/modals/components/radio_setting_item';

import {isMac} from 'utils/user_agent';

export enum AdvanceSettings {
    CTRL_SEND='ctrlSend',
    SEND_ON_CTRL_ENTER= 'send_on_ctrl_enter',
    CODE_BLOCK_CTRL_ENTER= 'code_block_ctrl_enter',
    FORMATTING='formatting',
    JOIN_LEAVE='join_leave',
    NAME_DISABLE_CLIENT_PLUGINS= 'disable_client_plugins',
    NAME_DISABLE_TELEMETRY= 'disable_telemetry',
    NAME_DISABLE_TYPING_MESSAGES= 'disable_typing_messages',
    UNREAD_SCROLL_POSITION= 'unread_scroll_position',
    SYNCED_DRAFTS_ARE_ALLOWED = 'synced_drafts_are_allowed'
}

export const enum OnAndOff {
    ON= 'on',
    OFF= 'off',
}

export const enum UnreadScrollPosition {
    'LEFT'= 'start_from_left_off',
    'NEWEST'= 'start_from_newest',
}

export const getCtrlSelectedOption = (sendOnCtrlEnter: string, codeBlockOnCtrlEnter: string) => {
    if (sendOnCtrlEnter === 'true' && codeBlockOnCtrlEnter === 'false') {
        return AdvanceSettings.SEND_ON_CTRL_ENTER;
    } else if (sendOnCtrlEnter === 'false' && codeBlockOnCtrlEnter === 'true') {
        return AdvanceSettings.CODE_BLOCK_CTRL_ENTER;
    }
    return OnAndOff.OFF;
};

export const getCtrlSendPreferenceValue = (value: string): {[AdvanceSettings.SEND_ON_CTRL_ENTER]: string;[AdvanceSettings.CODE_BLOCK_CTRL_ENTER]: string} => {
    if (value === AdvanceSettings.SEND_ON_CTRL_ENTER) {
        return {[AdvanceSettings.SEND_ON_CTRL_ENTER]: 'true', [AdvanceSettings.CODE_BLOCK_CTRL_ENTER]: 'false'};
    } else if (value === AdvanceSettings.CODE_BLOCK_CTRL_ENTER) {
        return {[AdvanceSettings.SEND_ON_CTRL_ENTER]: 'false', [AdvanceSettings.CODE_BLOCK_CTRL_ENTER]: 'true'};
    }
    return {[AdvanceSettings.SEND_ON_CTRL_ENTER]: 'false', [AdvanceSettings.CODE_BLOCK_CTRL_ENTER]: 'false'};
};

export const ctrlSendInputFieldData: FieldsetRadio = {
    options: [
        {
            dataTestId: `${AdvanceSettings.CTRL_SEND}-${AdvanceSettings.SEND_ON_CTRL_ENTER}`,
            title: (
                <FormattedMessage
                    id='user.settings.advance.onForAllMessages'
                    defaultMessage='On for all messages'
                />
            ),
            name: `${AdvanceSettings.CTRL_SEND}`,
            key: `${AdvanceSettings.CTRL_SEND}-${AdvanceSettings.SEND_ON_CTRL_ENTER}`,
            value: AdvanceSettings.SEND_ON_CTRL_ENTER,
        },
        {
            dataTestId: `${AdvanceSettings.CTRL_SEND}-${AdvanceSettings.CODE_BLOCK_CTRL_ENTER}`,
            title: (
                <FormattedMessage
                    id='user.settings.advance.onForCode'
                    defaultMessage='On only for code blocks starting with ```'
                />
            ),
            name: `${AdvanceSettings.CTRL_SEND}`,
            key: `${AdvanceSettings.CTRL_SEND}-${AdvanceSettings.CODE_BLOCK_CTRL_ENTER}`,
            value: AdvanceSettings.CODE_BLOCK_CTRL_ENTER,
        },
        {
            dataTestId: `${AdvanceSettings.CTRL_SEND}-${OnAndOff.OFF}`,
            title: (
                <FormattedMessage
                    id='user.settings.advance.off'
                    defaultMessage='Off'
                />
            ),
            name: `${AdvanceSettings.CTRL_SEND}`,
            key: `${AdvanceSettings.CTRL_SEND}-${OnAndOff.OFF}`,
            value: OnAndOff.OFF,
        },
    ],
};

export const FormattingInputFieldData: FieldsetRadio = {
    options: [
        {
            dataTestId: `${AdvanceSettings.FORMATTING}-${OnAndOff.ON}`,
            title: (
                <FormattedMessage
                    id='ser.settings.advance.on'
                    defaultMessage='On'
                />
            ),
            name: `${AdvanceSettings.FORMATTING}`,
            key: `${AdvanceSettings.FORMATTING}-${OnAndOff.ON}`,
            value: OnAndOff.ON,
        },
        {
            dataTestId: `${AdvanceSettings.FORMATTING}-${OnAndOff.OFF}`,
            title: (
                <FormattedMessage
                    id='user.settings.advance.off'
                    defaultMessage='Off'
                />
            ),
            name: `${AdvanceSettings.FORMATTING}`,
            key: `${AdvanceSettings.FORMATTING}-${OnAndOff.OFF}`,
            value: OnAndOff.OFF,
        },
    ],
};

export const JoinLeaveInputFieldData: FieldsetRadio = {
    options: [
        {
            dataTestId: `${AdvanceSettings.JOIN_LEAVE}-${OnAndOff.ON}`,
            title: (
                <FormattedMessage
                    id='user.settings.advance.on'
                    defaultMessage='On'
                />
            ),
            name: `${AdvanceSettings.JOIN_LEAVE}`,
            key: `${AdvanceSettings.JOIN_LEAVE}-${OnAndOff.ON}`,
            value: OnAndOff.ON,
        },
        {
            dataTestId: `${AdvanceSettings.JOIN_LEAVE}-${OnAndOff.OFF}`,
            title: (
                <FormattedMessage
                    id='user.settings.advance.off'
                    defaultMessage='Off'
                />
            ),
            name: `${AdvanceSettings.JOIN_LEAVE}`,
            key: `${AdvanceSettings.JOIN_LEAVE}-${OnAndOff.OFF}`,
            value: OnAndOff.OFF,
        },
    ],
};

export const SyncedDraftInputFieldData: FieldsetRadio = {
    options: [
        {
            dataTestId: `${AdvanceSettings.SYNCED_DRAFTS_ARE_ALLOWED}-${OnAndOff.ON}`,
            title: (
                <FormattedMessage
                    id='user.settings.advance.on'
                    defaultMessage='On'
                />
            ),
            name: `${AdvanceSettings.SYNCED_DRAFTS_ARE_ALLOWED}`,
            key: `${AdvanceSettings.SYNCED_DRAFTS_ARE_ALLOWED}-${OnAndOff.ON}`,
            value: OnAndOff.ON,
        },
        {
            dataTestId: `${AdvanceSettings.SYNCED_DRAFTS_ARE_ALLOWED}-${OnAndOff.OFF}`,
            title: (
                <FormattedMessage
                    id='user.settings.advance.off'
                    defaultMessage='Off'
                />
            ),
            name: `${AdvanceSettings.SYNCED_DRAFTS_ARE_ALLOWED}`,
            key: `${AdvanceSettings.SYNCED_DRAFTS_ARE_ALLOWED}-${OnAndOff.OFF}`,
            value: OnAndOff.OFF,
        },
    ],
};

export const UnreadScrollPositionInputFieldData: FieldsetRadio = {
    options: [
        {
            dataTestId: `${AdvanceSettings.UNREAD_SCROLL_POSITION}-${OnAndOff.ON}`,
            title: (
                <FormattedMessage
                    id='user.settings.advance.startFromLeftOff'
                    defaultMessage='Start me where I left off'
                />
            ),
            name: `${AdvanceSettings.UNREAD_SCROLL_POSITION}`,
            key: `${AdvanceSettings.UNREAD_SCROLL_POSITION}-${OnAndOff.ON}`,
            value: UnreadScrollPosition.LEFT,
        },
        {
            dataTestId: `${AdvanceSettings.UNREAD_SCROLL_POSITION}-${OnAndOff.OFF}`,
            title: (
                <FormattedMessage
                    id='user.settings.advance.startFromNewest'
                    defaultMessage='Start me at the newest message'
                />
            ),
            name: `${AdvanceSettings.UNREAD_SCROLL_POSITION}`,
            key: `${AdvanceSettings.UNREAD_SCROLL_POSITION}-${OnAndOff.OFF}`,
            value: UnreadScrollPosition.NEWEST,
        },
    ],
};

export const getCtrlSendText = () => {
    const description = {
        default: defineMessage({
            id: 'user.settings.advance.sendDesc',
            defaultMessage: 'When enabled, CTRL + ENTER will send the message and ENTER inserts a new line.',
        }),
        mac: defineMessage({
            id: 'user.settings.advance.sendDesc.mac',
            defaultMessage: 'When enabled, ⌘ + ENTER will send the message and ENTER inserts a new line.',
        }),
    };
    const title = {
        default: defineMessage({
            id: 'user.settings.advance.sendTitle',
            defaultMessage: 'Send Messages on CTRL+ENTER',
        }),
        mac: ({
            id: 'user.settings.advance.sendTitle.mac',
            defaultMessage: 'Send Messages on ⌘+ENTER',
        }),
    };
    if (isMac()) {
        return {
            ctrlSendTitle: title.mac,
            ctrlSendDesc: description.mac,
        };
    }
    return {
        ctrlSendTitle: title.default,
        ctrlSendDesc: description.default,
    };
};

export const FormattingSectionTitle = defineMessage({
    id: 'user.settings.advance.formattingTitle',
    defaultMessage: 'Enable Post Formatting',
});

export const FormattingSectionDesc = defineMessage({
    id: 'user.settings.advance.formattingDesc',
    defaultMessage: 'If enabled, posts will be formatted to create links, show emoji, style the text, and add line breaks. By default, this setting is enabled.',
});

export const JoinLeaveSectionTitle = defineMessage({
    id: 'user.settings.advance.joinLeaveTitle',
    defaultMessage: 'Join/leave messages',
});

export const JoinLeaveSectionDesc = defineMessage({
    id: 'user.settings.advance.joinLeaveDesc',
    defaultMessage: 'When enabled, system messages display when users join or leave Mattermost channels.',
});

export const UnreadScrollPositionSectionTitle = defineMessage({
    id: 'user.settings.advance.unreadScrollPositionTitle',
    defaultMessage: 'Scroll position when viewing an unread channel',
});

export const UnreadScrollPositionSectionDesc = defineMessage({
    id: 'user.settings.advance.unreadScrollPositionDesc',
    defaultMessage: 'Choose your scroll position when you view an unread channel. Channels will always be marked as read when viewed.',
});

export const SyncedDraftSectionTitle = defineMessage({
    id: 'user.settings.advance.syncDrafts.Title',
    defaultMessage: 'Allow message drafts to sync with the server',
});

export const SyncedDraftSectionDesc = defineMessage({
    id: 'user.settings.advance.syncDrafts.Desc',
    defaultMessage: 'When enabled, message drafts are synced with the server so they can be accessed from any device. When disabled, message drafts are only saved locally on the device where they are composed.',
});

export const PerformanceDebuggingSectionTitle = defineMessage({
    id: 'user.settings.advance.performance.title',
    defaultMessage: 'Performance Debugging',
});

export const PerformanceDebuggingSectionDesc = defineMessage({
    id: 'user.settings.advance.performance.info1',
    defaultMessage: 'You may enable these settings temporarily to help isolate performance issues while debugging. We don\'t recommend leaving these settings enabled for an extended period of time as they can negatively impact your user experience. You may need to refresh the page before these settings take effect.',
});

export const DebuggingPluginInputFieldData: FieldsetCheckbox = {
    name: 'plugin',
    dataTestId: 'plugin',
};

export const DebuggingTelemetryInputFieldData: FieldsetCheckbox = {

    name: 'telemetry',
    dataTestId: 'telemetry',
};

export const DebuggingTypingInputFieldData: FieldsetCheckbox = {
    name: 'typing',
    dataTestId: 'typing',
};
