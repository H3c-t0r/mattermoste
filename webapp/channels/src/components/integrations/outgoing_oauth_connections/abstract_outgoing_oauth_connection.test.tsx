// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import type {OutgoingOAuthConnection} from '@mattermost/types/integrations';

import AbstractOutgoingOAuthConnection from 'components/integrations/outgoing_oauth_connections/abstract_outgoing_oauth_connection';

import {TestHelper} from 'utils/test_helper';

describe('components/integrations/AbstractOutgoingOAuthConnection', () => {
    const header = {id: 'Header', defaultMessage: 'Header'};
    const footer = {id: 'Footer', defaultMessage: 'Footer'};
    const loading = {id: 'Loading', defaultMessage: 'Loading'};
    const initialConnection: OutgoingOAuthConnection = {
        id: 'facxd9wpzpbpfp8pad78xj75pr',
        name: 'testConnection',
        client_secret: '',
        client_id: 'clientid',
        create_at: 1501365458934,
        creator_id: '88oybd1dwfdoxpkpw1h5kpbyco',
        update_at: 1501365458934,
        grant_type: 'client_credentials',
        oauth_token_url: 'https://token.com',
        audiences: ['https://aud.com'],
    };

    const action = jest.fn().mockImplementation(
        () => {
            return new Promise<void>((resolve) => {
                process.nextTick(() => resolve());
            });
        },
    );

    const team = TestHelper.getTeamMock({name: 'test', id: initialConnection.id});

    const baseProps: React.ComponentProps<typeof AbstractOutgoingOAuthConnection> = {
        team,
        header,
        footer,
        loading,
        renderExtra: <div>{'renderExtra'}</div>,
        serverError: '',
        initialConnection,
        action: jest.fn(),
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <AbstractOutgoingOAuthConnection {...baseProps}/>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, displays client error', () => {
        const newServerError = 'serverError';
        const props = {...baseProps, serverError: newServerError};
        const wrapper = shallow(
            <AbstractOutgoingOAuthConnection {...props}/>,
        );

        wrapper.find('#audienceUrls').simulate('change', {target: {value: ''}});
        wrapper.find('.btn-primary').simulate('click', {preventDefault() {
            return jest.fn();
        }});

        expect(action).not.toBeCalled();
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('FormError').exists()).toBe(true);
    });

    test('should call action function', () => {
        const props = {...baseProps, action};
        const wrapper = shallow(
            <AbstractOutgoingOAuthConnection {...props}/>,
        );

        wrapper.find('#name').simulate('change', {target: {value: 'name'}});
        wrapper.find('.btn-primary').simulate('click', {preventDefault() {
            return jest.fn();
        }});

        expect(action).toBeCalled();
    });
});
