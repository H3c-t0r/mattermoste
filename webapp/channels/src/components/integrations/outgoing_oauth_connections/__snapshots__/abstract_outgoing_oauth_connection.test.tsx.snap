// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/integrations/AbstractOutgoingOAuthConnection should match snapshot 1`] = `
<div
  className="backstage-content"
>
  <BackstageHeader>
    <Link
      to="/test/integrations/outgoing-oauth2-connections"
    >
      <MemoizedFormattedMessage
        defaultMessage="Outgoing OAuth Connections"
        id="add_outgoing_oauth_connection.header"
      />
    </Link>
    <MemoizedFormattedMessage
      defaultMessage="Header"
      id="Header"
    />
  </BackstageHeader>
  <div
    className="backstage-form"
  >
    <form
      className="form-horizontal"
    >
      <div
        className="form-group"
      >
        <label
          className="control-label col-sm-4"
          htmlFor="name"
        >
          <MemoizedFormattedMessage
            defaultMessage="Name"
            id="add_outgoing_oauth_connection.name.label"
          />
        </label>
        <div
          className="col-md-5 col-sm-8"
        >
          <input
            className="form-control"
            id="name"
            maxLength={64}
            onChange={[Function]}
            type="text"
            value="testConnection"
          />
          <div
            className="form__help"
          >
            <MemoizedFormattedMessage
              defaultMessage="Specify the name for your OAuth connection."
              id="add_outgoing_oauth_connection.name.help"
            />
          </div>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label col-sm-4"
          htmlFor="client_id"
        >
          <MemoizedFormattedMessage
            defaultMessage="Client ID"
            id="add_outgoing_oauth_connection.client_id.label"
          />
        </label>
        <div
          className="col-md-5 col-sm-8"
        >
          <input
            className="form-control"
            id="client_id"
            maxLength={64}
            onChange={[Function]}
            type="text"
            value="clientid"
          />
          <div
            className="form__help"
          >
            <MemoizedFormattedMessage
              defaultMessage="Specify the Client ID for your OAuth connection."
              id="add_outgoing_oauth_connection.client_id.help"
            />
          </div>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label col-sm-4"
          htmlFor="client_secret"
        >
          <MemoizedFormattedMessage
            defaultMessage="Client Secret"
            id="add_outgoing_oauth_connection.client_secret.label"
          />
        </label>
        <div
          className="col-md-5 col-sm-8"
        >
          <input
            className="form-control disabled"
            disabled={true}
            id="client_secret"
            type="text"
            value="****************"
          />
          <span
            className="outgoing-oauth-connections-edit-secret"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "position": "absolute",
                "right": "16px",
                "top": "8px",
              }
            }
          >
            <i
              className="icon icon-pencil-outline"
            />
          </span>
          <div
            className="form__help"
          >
            <MemoizedFormattedMessage
              defaultMessage="Specify the Client Secret for your OAuth connection."
              id="add_outgoing_oauth_connection.client_secret.help"
            />
          </div>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label col-sm-4"
          htmlFor="oauth_token_url"
        >
          <MemoizedFormattedMessage
            defaultMessage="OAuth Token URL"
            id="add_outgoing_oauth_connection.oauth_token_url.label"
          />
        </label>
        <div
          className="col-md-5 col-sm-8"
        >
          <input
            className="form-control"
            id="token_url"
            maxLength={64}
            onChange={[Function]}
            type="text"
            value="https://token.com"
          />
          <div
            className="form__help"
          >
            <MemoizedFormattedMessage
              defaultMessage="Specify the OAuth Token URL for your OAuth connection."
              id="add_outgoing_oauth_connection.oauth_token_url.help"
            />
          </div>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label col-sm-4"
          htmlFor="audienceUrls"
        >
          <MemoizedFormattedMessage
            defaultMessage="Audience URLs (One Per Line)"
            id="add_outgoing_oauth_connection.audienceUrls.label"
          />
        </label>
        <div
          className="col-md-5 col-sm-8"
        >
          <textarea
            className="form-control"
            id="audienceUrls"
            maxLength={1024}
            onChange={[Function]}
            rows={3}
            value="https://aud.com"
          />
          <div
            className="form__help"
          >
            <MemoizedFormattedMessage
              defaultMessage="The audience URIs which will receive requests with the OAuth token. Must be a valid URL and start with http:// or https://."
              id="add_outgoing_oauth_connection.audienceUrls.help"
            />
          </div>
        </div>
      </div>
      <div
        className="backstage-form__footer"
      >
        <FormError
          error={null}
          errors={
            Array [
              "",
              "",
            ]
          }
          type="backstage"
        />
        <Link
          className="btn btn-tertiary"
          to="/test/integrations/outgoing-oauth2-connections"
        >
          <MemoizedFormattedMessage
            defaultMessage="Cancel"
            id="add_outgoing_oauth_connection.cancel"
          />
        </Link>
        <Memo(SpinnerButton)
          className="btn btn-primary"
          id="saveOauthApp"
          onClick={[Function]}
          spinning={false}
          spinningText="Loading"
          type="submit"
        >
          <MemoizedFormattedMessage
            defaultMessage="Footer"
            id="Footer"
          />
        </Memo(SpinnerButton)>
        <div>
          renderExtra
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`components/integrations/AbstractOutgoingOAuthConnection should match snapshot, displays client error 1`] = `
<div
  className="backstage-content"
>
  <BackstageHeader>
    <Link
      to="/test/integrations/outgoing-oauth2-connections"
    >
      <MemoizedFormattedMessage
        defaultMessage="Outgoing OAuth Connections"
        id="add_outgoing_oauth_connection.header"
      />
    </Link>
    <MemoizedFormattedMessage
      defaultMessage="Header"
      id="Header"
    />
  </BackstageHeader>
  <div
    className="backstage-form"
  >
    <form
      className="form-horizontal"
    >
      <div
        className="form-group"
      >
        <label
          className="control-label col-sm-4"
          htmlFor="name"
        >
          <MemoizedFormattedMessage
            defaultMessage="Name"
            id="add_outgoing_oauth_connection.name.label"
          />
        </label>
        <div
          className="col-md-5 col-sm-8"
        >
          <input
            className="form-control"
            id="name"
            maxLength={64}
            onChange={[Function]}
            type="text"
            value="testConnection"
          />
          <div
            className="form__help"
          >
            <MemoizedFormattedMessage
              defaultMessage="Specify the name for your OAuth connection."
              id="add_outgoing_oauth_connection.name.help"
            />
          </div>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label col-sm-4"
          htmlFor="client_id"
        >
          <MemoizedFormattedMessage
            defaultMessage="Client ID"
            id="add_outgoing_oauth_connection.client_id.label"
          />
        </label>
        <div
          className="col-md-5 col-sm-8"
        >
          <input
            className="form-control"
            id="client_id"
            maxLength={64}
            onChange={[Function]}
            type="text"
            value="clientid"
          />
          <div
            className="form__help"
          >
            <MemoizedFormattedMessage
              defaultMessage="Specify the Client ID for your OAuth connection."
              id="add_outgoing_oauth_connection.client_id.help"
            />
          </div>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label col-sm-4"
          htmlFor="client_secret"
        >
          <MemoizedFormattedMessage
            defaultMessage="Client Secret"
            id="add_outgoing_oauth_connection.client_secret.label"
          />
        </label>
        <div
          className="col-md-5 col-sm-8"
        >
          <input
            className="form-control disabled"
            disabled={true}
            id="client_secret"
            type="text"
            value="****************"
          />
          <span
            className="outgoing-oauth-connections-edit-secret"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "position": "absolute",
                "right": "16px",
                "top": "8px",
              }
            }
          >
            <i
              className="icon icon-pencil-outline"
            />
          </span>
          <div
            className="form__help"
          >
            <MemoizedFormattedMessage
              defaultMessage="Specify the Client Secret for your OAuth connection."
              id="add_outgoing_oauth_connection.client_secret.help"
            />
          </div>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label col-sm-4"
          htmlFor="oauth_token_url"
        >
          <MemoizedFormattedMessage
            defaultMessage="OAuth Token URL"
            id="add_outgoing_oauth_connection.oauth_token_url.label"
          />
        </label>
        <div
          className="col-md-5 col-sm-8"
        >
          <input
            className="form-control"
            id="token_url"
            maxLength={64}
            onChange={[Function]}
            type="text"
            value="https://token.com"
          />
          <div
            className="form__help"
          >
            <MemoizedFormattedMessage
              defaultMessage="Specify the OAuth Token URL for your OAuth connection."
              id="add_outgoing_oauth_connection.oauth_token_url.help"
            />
          </div>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label col-sm-4"
          htmlFor="audienceUrls"
        >
          <MemoizedFormattedMessage
            defaultMessage="Audience URLs (One Per Line)"
            id="add_outgoing_oauth_connection.audienceUrls.label"
          />
        </label>
        <div
          className="col-md-5 col-sm-8"
        >
          <textarea
            className="form-control"
            id="audienceUrls"
            maxLength={1024}
            onChange={[Function]}
            rows={3}
            value=""
          />
          <div
            className="form__help"
          >
            <MemoizedFormattedMessage
              defaultMessage="The audience URIs which will receive requests with the OAuth token. Must be a valid URL and start with http:// or https://."
              id="add_outgoing_oauth_connection.audienceUrls.help"
            />
          </div>
        </div>
      </div>
      <div
        className="backstage-form__footer"
      >
        <FormError
          error={null}
          errors={
            Array [
              "serverError",
              <Memo(MemoizedFormattedMessage)
                defaultMessage="One or more audience URLs are required."
                id="add_outgoing_oauth_connection.audienceUrls.required"
              />,
            ]
          }
          type="backstage"
        />
        <Link
          className="btn btn-tertiary"
          to="/test/integrations/outgoing-oauth2-connections"
        >
          <MemoizedFormattedMessage
            defaultMessage="Cancel"
            id="add_outgoing_oauth_connection.cancel"
          />
        </Link>
        <Memo(SpinnerButton)
          className="btn btn-primary"
          id="saveOauthApp"
          onClick={[Function]}
          spinning={false}
          spinningText="Loading"
          type="submit"
        >
          <MemoizedFormattedMessage
            defaultMessage="Footer"
            id="Footer"
          />
        </Memo(SpinnerButton)>
        <div>
          renderExtra
        </div>
      </div>
    </form>
  </div>
</div>
`;
